### PR TITLE
Add back in support for Dictionary Class Weights

### PR DESCRIPTION
**Pull request by connor-mccorm**

### CODE REVIEW

File: ludwig/schema/features/loss/loss.py [hunk 1]
```diff
@@ -310,10 +313,13 @@ class SequenceSoftmaxCrossEntropyLossConfig(BaseLossConfig):
         description="Type of loss.",
     )
 
-    class_weights: Union[List[float], float, None] = schema_utils.List(
-        list_type=float,
+    class_weights: Union[List[float], float, dict, None] = schema_utils.OneOfOptionsField(
```
Comment (ksbrar) [inline ludwig/schema/features/loss/loss.py:316]: nit: type annotation does not match the field options

File: ludwig/schema/features/loss/loss.py [hunk 2]
```diff
@@ -258,10 +258,13 @@ class SoftmaxCrossEntropyLossConfig(BaseLossConfig):
...
```
Comment (ksbrar) [inline ludwig/schema/features/loss/loss.py:261]: nit: type annotation does not match the field options

File: ludwig/schema/features/loss/loss.py [hunk 3]
```diff
@@ -368,10 +374,13 @@ class SigmoidCrossEntropyLossConfig(BaseLossConfig):
...
```
Comment (ksbrar) [inline ludwig/schema/features/loss/loss.py:377]: nit: type annotation does not match the field options

Comment (connor-mccorm) [reply ludwig/schema/features/loss/loss.py]: Good catch!

File: tests/ludwig/utils/test_backward_compatibility.py [hunk 4]
```diff
@@ -542,43 +540,6 @@ def test_update_increase_batch_size_on_plateau_max():
...
-                "name": "input_feature_1",
-                "type": "category",
-            }
-        ],
-        "output_features": [
-            {"name": "output_feature_1", "type": "category", "loss": {"class_weights": None}},
-        ],
-    }
-
-    upgraded_config = upgrade_config_dict_to_latest_version(old_config)
```
Comment (ksbrar) [inline tests/ludwig/utils/test_backward_compatibility.py:570]: I think we'd still wanna test the upgrade, no?

Comment (connor-mccorm) [reply tests/ludwig/utils/test_backward_compatibility.py:570]: Ah good looks, I removed this because I removed the backwards compatibility step before, but added it back in and changed it per another issue. Will add this back in today!

File: tests/ludwig/modules/test_loss_modules.py [hunk 5]
```diff
@@ -144,3 +149,151 @@ def test_huber_loss(
 def test_corn_loss(preds: torch.Tensor, target: torch.Tensor, output: torch.Tensor):
     loss = loss_modules.CORNLoss(CORNLossConfig())
     assert torch.isclose(loss(preds, target), output, rtol=0.0001)
+
+
+def test_dict_class_weights_category():
```
Comment (ksbrar) [inline tests/ludwig/modules/test_loss_modules.py:154]: Do we have a test of class weights as a list?

**Review by ksbrar — APPROVED** [review]

File: tests/ludwig/modules/test_loss_modules.py (hunk 5 above)
Comment (connor-mccorm) [reply tests/ludwig/modules/test_loss_modules.py:154]: Ya we have `test_experiment_multiclass_with_class_weights` in `test_experiment.py` which uses list class weights and checks the expected outcome.
